handlers/public.go

```diff
@@ -0,0 +1,208 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payments.api.ch.gov.uk/config"
+	"github.com/companieshouse/payments.api.ch.gov.uk/data"
+
+	"github.com/gorilla/pat"
+)
+
+type createPaymentResource struct {
+	RedirectURI string `json:"redirect_uri"`
+	Reference   string `json:"reference"`
+	Resource    string `json:"resource"`
+	State       string `json:"state"`
+}
+
+// PaymentResource is the payment details to be stored in the DB and returned in the response
+type PaymentResource struct {
+	//	ID                      string    `json:"id"                                  bson:"_id"`                                 // TODO implement
+	Amount                  string    `json:"amount"                              bson:"amount"`                              // TODO implement
+	AvailablePaymentMethods []string  `json:"available_payment_methods,omitempty" bson:"available_payment_methods,omitempty"` // TODO implement
+	CompletedAt             time.Time `json:"completed_at,omitempty"              bson:"completed_at,omitempty"`              // TODO implement
+	CreatedAt               time.Time `json:"created_at,omitempty"                bson:"created_at,omitempty"`
+	CreatedBy               CreatedBy `json:"created_by"                          bson:"created_by"`
+	Description             string    `json:"description"                         bson:"description"`              // TODO implement
+	Links                   Links     `json:"links"                               bson:"links"`                    // TODO implement
+	PaymentMethod           string    `json:"payment_method,omitempty"            bson:"payment_method,omitempty"` // TODO implement
+	Reference               string    `json:"reference,omitempty"                 bson:"reference,omitempty"`      // TODO implement
+	Status                  string    `json:"status"                              bson:"status"`                   // TODO implement
```

Comment:
Why are there lots of TODOs on this struct? Can you add one TODO referencing the story which explains what needs to be implemented please? It is not clear what's missing and would be difficult to figure out if the TODOs have been satisfied.